src/LinuxTracing/UprobesUnwindingVisitorTest.cpp

```diff
@@ -265,8 +319,302 @@ TEST_F(UprobesUnwindingVisitorTest,
+      .time = 15,
+      .stream_id = 12,
+      .cpu = 0,
+      .res = 0,
+  };
+  event.ring_buffer_record.sample_id = sample_id;
+  event.ips = callchain;
+
+  EXPECT_CALL(maps_, Find).WillRepeatedly(Return(&kTargetMapInfo));
+  EXPECT_CALL(return_address_manager_, PatchCallchain).WillRepeatedly(Return(false));
+
+  std::vector<unwindstack::FrameData> libunwindstack_callstack;
+  libunwindstack_callstack.push_back(kFrame1);
+  libunwindstack_callstack.push_back(kFrame2);
+
+  EXPECT_CALL(unwinder_, Unwind(kPid, nullptr, _, _, kStackSize))
+      .WillRepeatedly(Return(libunwindstack_callstack));
+
+  EXPECT_CALL(listener_, OnCallstackSample).Times(0);
+
+  std::atomic<uint64_t> unwinding_errors = 0;
+  std::atomic<uint64_t> discarded_samples_in_uretprobes_counter = 0;
+  visitor_->SetUnwindErrorsAndDiscardedSamplesCounters(&unwinding_errors,
+                                                       &discarded_samples_in_uretprobes_counter);
+
+  visitor_->visit(&event);
+
+  EXPECT_EQ(unwinding_errors, 1);
+  EXPECT_EQ(discarded_samples_in_uretprobes_counter, 0);
+}
+
+TEST_F(UprobesUnwindingVisitorTest, VisitCallchainSampleInsideUprobeCodeDiscardEvents) {
+  constexpr uint32_t kPid = 10;
+  constexpr uint64_t kStackSize = 13;
+
+  std::vector<uint64_t> callchain;
+  callchain.push_back(kKernelAddress);
+  callchain.push_back(kUprobesMapsStart);
+  // Increment by one as the return address is the next address.
+  callchain.push_back(kTargetAddress2 + 1);
+  callchain.push_back(kTargetAddress3 + 1);
+
+  CallchainSamplePerfEvent event{callchain.size(), kStackSize};
+  perf_event_sample_id_tid_time_streamid_cpu sample_id{
+      .pid = kPid,
+      .tid = 11,
+      .time = 15,
+      .stream_id = 12,
+      .cpu = 0,
+      .res = 0,
+  };
+  event.ring_buffer_record.sample_id = sample_id;
+  event.ips = callchain;
+
+  EXPECT_CALL(maps_, Find(_)).WillRepeatedly(Return(&kTargetMapInfo));
+  EXPECT_CALL(maps_, Find(kUprobesMapsStart)).WillRepeatedly(Return(&kUprobesMapInfo));
+  EXPECT_CALL(return_address_manager_, PatchCallchain).WillRepeatedly(Return(true));
+
+  std::vector<unwindstack::FrameData> libunwindstack_callstack;
+  libunwindstack_callstack.push_back(kFrame1);
+  libunwindstack_callstack.push_back(kFrame2);
+
+  EXPECT_CALL(unwinder_, Unwind(kPid, nullptr, _, _, kStackSize))
+      .WillRepeatedly(Return(libunwindstack_callstack));
+
+  EXPECT_CALL(listener_, OnCallstackSample).Times(0);
+
+  std::atomic<uint64_t> unwinding_errors = 0;
+  std::atomic<uint64_t> discarded_samples_in_uretprobes_counter = 0;
+  visitor_->SetUnwindErrorsAndDiscardedSamplesCounters(&unwinding_errors,
+                                                       &discarded_samples_in_uretprobes_counter);
+
+  visitor_->visit(&event);
+
+  EXPECT_EQ(unwinding_errors, 0);
+  EXPECT_EQ(discarded_samples_in_uretprobes_counter, 1);
+}
+
+TEST_F(UprobesUnwindingVisitorTest,
+       DISABLED_VisitLeafCallOptimizedCallchainSampleWithoutUprobesEmitsEvents) {
```

Comment:
Add a comment why this is disabled, possibly with a TODO linking a bug. Or just consider adding this in another change.

Reply:
Done.